amalgamate.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+set -euox pipefail
+
+name=cfs
+
+rust_dir=${name}_rust_amalgamated
+
+# export CC=clang
+
+create-compile-commands() {
+    cp cfe/cmake/Makefile.sample Makefile
+    git apply Makefile.diff
+
+    rm -rf sample_defs
+    cp -r cfe/cmake/sample_defs sample_defs
+
+    # rm -rf build
+    rm -f compile_commands.json
+    export PREP_OPTS="-DCMAKE_EXPORT_COMPILE_COMMANDS=On"
+    make \
+        BUILDTYPE=debug \
+        SIMULATION=native \
+        ENABLE_UNIT_TESTS=false \
+        OMIT_DEPRECATED=false \
+        prep
+    ln build/native/default_cpu1/compile_commands.json .
+}
+
+create-compile-commands
+
+make # creates some config files we need
+
+amalgamated_dir="amalgamated"
+errors_log_path=${amalgamated_dir}/errors.log
+files_with_errors_path=${amalgamated_dir}/files_with_errors.txt
+
+rm -rf ${amalgamated_dir}
+mkdir ${amalgamated_dir}
+
+# ./amalgamate.mjs cfs | \
+#     ("${SHELL}" -euox pipefail || true) \
+#     |& rg 'error: redefinition of ‘([^’]*)’' --only-matching --replace '$1' \
+#     > ${amalgamated_dir}/redefined_identifiers.txt
+
+# # skip files with errors
+# rm -f ${errors_log_path} ${files_with_errors_path}
+# touch ${errors_log_path} ${files_with_errors_path}
+
+# # Do a fixpoint loop until we get to 0 errors.
+# # gcc does this in 1-2 iterations, clang takes forever.
+# while : ; do
+#     prev_num_files_with_errors=$(wc -l < ${files_with_errors_path})
+#     ./amalgamate.mjs cfs | ("${SHELL}" -euox pipefail || true) &>> ${errors_log_path}
+#     rg '([^:]+):[0-9]+:[0-9]+:' --only-matching --replace '$1' < ${errors_log_path} > ${files_with_errors_path}
+#     num_files_with_errors=$(wc -l < ${files_with_errors_path})
+#     if [[ ${num_files_with_errors} -eq ${prev_num_files_with_errors} ]]; then
+#         break
+#     fi
+# done
+
+# all errors should be gone now; we reached a fixpoint of 0 new errors above
+# echo skipped all errors
+./amalgamate.mjs ${name} | "${SHELL}" -euox pipefail
+
+# binary_name=${name}_amalgamated
+binary_name=${name}
+c2rust transpile \
+    --overwrite-existing \
+    --emit-build-files \
+    --binary ${binary_name} \
+    --output-dir ${rust_dir} ${amalgamated_dir}/compile_commands.json \
+
+rm -rf ${rust_dir}.old
+mv ${rust_dir} ${rust_dir}.old
+cargo new ${rust_dir}
+(cd "${rust_dir}"
+    cargo add libc
+    cargo add c2rust-bitfields
+    cargo add memoffset
+    cargo add f128
+
+    # mv ../${rust_dir}.old/build.rs . # Don't need it.
```

Comment:
Why not delete this line?

Reply:
It was from [`lighttpd-rust/amalgamate.mjs`](https://github.com/immunant/lighttpd-rust/blob/385ca548101242cfb8f8bc61533f1073042b5036/amalgamate.mjs).  And since I did things very quick and dirty, I wanted to leave in stuff I may want to re-enable in the future if I encountered problems or something.